flang/runtime/CMakeLists.txt

```diff
@@ -103,7 +103,13 @@ append(${NO_LTO_FLAGS} CMAKE_CXX_FLAGS)
 add_definitions(-U_GLIBCXX_ASSERTIONS)
 add_definitions(-U_LIBCPP_ENABLE_ASSERTIONS)
 
-add_subdirectory(Float128Math)
+set(FLANG_RUNTIME_ENABLE_F128 On CACHE BOOL
```

Comment:
We check for F128 in other places too. See https://github.com/llvm/llvm-project/pull/106957

Reply:
I was sort of just trying to get this as a quick fix for building the runtime. I think fixing up all the warnings etc might be a bit more effort.

Reply:
I guess it is okay as an engineering temporary switch, but, in general, uncoordinated settings for float128 math for the compiler driver and runtime may result in completely unsuable driver, e.g. if Float128Math is packaged as a separate library (rather than as part of FortranRuntime) then the driver will try to link it always, so it has to be built during the runtime build.  As I uderstand this is not your case, and Float128Math is a part of FortranRuntime (except that it does not build at all due to missing `ynl`).

If `ynl` is the only missing piece in those configurations, you can try to ifdef `DEFINE_SIMPLE_ALIAS(Yn, ynl)` in `math-entries.h` for those targets.  The library will compile, but there will be a runtime error if a program uses `bessel_yn` for `real(16)`.

Reply:
As I understand it, we currently check if the libc has float 128 support by checking if `LDBL_MANT_DIG` has a certain value. The issue I guess is that that's sufficient for checking that libc will provide float 128 support for the _standard_ libc functions, but `ynl` is a glibc-only function. 

I think it'd be nice if we could have this support everywhere that does have float128 not just glibc; could we maybe re-use the implementation from llvm/libc++ somehow without introducing a runtime dependency on the system c++ library? And possibly the same for any other glibc-specific functions that might be being used (I haven't checked for others)?

Reply:
Or I suppose if there are no targets on which we expect float128 to be available but not a C++ standard library, could we introduce a dependency on the system c++ library just for float128 support?

Reply:
Maybe we should use `#if defined(__GLIBC__) && defined(_GNU_SOURCE)` for defining the simple alias for `ynl`.

Regarding adding dependency on the C++ library, I think we want to avoid this for `FortranRuntime`, in general.  When float128 support is enabled via `libc`, the float128 math functions are actually a part of `FortranRuntime` and so using libc++ would break this "rule".

Reply:
I'll try that as a fix instead and see what happens. If it works I can open another PR and close this one. `ynl` might not be the only glibc-specific function though.